features/checkout/completing_cart_in_any_order.feature

```diff
@@ -0,0 +1,82 @@
+@checkout
+Feature: Add possibility to perform checkout in any order
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "PHP Mascot" priced at "$15.00"
+        And the store has a product "Sylius T-Shirt" priced at "$0.00"
+        And the store has a product "Toyota Kata eBook" priced at "$50.00"
+        And this product does not require shipping
+        And the store has a product "Sylius eBook" priced at "$0.00"
+        And this product does not require shipping
+        And the store has "SHL" shipping method with "$5.00" fee
+        And the store allows paying "offline"
+        And I am a logged in customer
+        And I use "async" checkout type
```

Comment:
IMHO this is a technical detail, all checkouts made by API should work in any order of steps